src/resources/Validator.js

```diff
@@ -343,6 +343,14 @@ class Validator {
       for (const path in context.scope.fetched) {
         _.unset(submission.data, path);
       }
+
+      // Now that the validation is complete, we need to remove null values from the submission so that they are not
+      // returned in the response body
+      for (const submissionKey in submission.data) {
+        if (submission.data[submissionKey] === null) {
+          delete submission.data[submissionKey];
+        }
+      }
```

Comment:
This might need to be a "falsey" check instead of just null, in case we want to filter out undefined and empty strings.
@brendanbond thoughts?
```suggestion
      for (const submissionKey in submission.data) {
        if (!submission.data[submissionKey]) {
          delete submission.data[submissionKey];
        }
      }
```

Reply:
My concern with `(!submission.data[submissionKey])` would be that if a submission data had for example a 0
`number: 0`
Then this would be falsey and remove it from the submission data.

Reply:
This is not where this change should occur. The ticket specifies `tag` components, and is an effort to maintain backwards compatibility with the 8.x release rather than a blanket effort to remove null values from submissions. This should go in the `normalize` processor of @formio/core